### PR TITLE
Example JSON model parser and Schema.

### PR DESCRIPTION
**Pull request by trivialfis**
* Add schema.
* Add an example model parser.
* Use `jsonschema` package to validate the schema and generated model.

### CODE REVIEW

**Comment by trivialfis** [comment]
@hcho3 @KOLANICH I merged the schema to here.  The display is not so nice but should suffice for now.  We can move all discussion to main XGBoost repo for future improvement.

**Comment by trivialfis** [comment]
Note to myself: add a ref to demo.

File: demo/json-model/json_parser.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            nid = stacks.pop()
+
+            node['node id'] = nid
+            node['gain'] = self.loss_change(nid)
+            node['cover'] = self.sum_hessian(nid)
+            nodes.append(node)
+
+            if not self.is_leaf(nid) and not self.is_deleted(nid):
+                left = self.left_child(nid)
+                rigth = self.right_child(nid)
```
Comment (hcho3) [inline demo/json-model/json_parser.py]:
```suggestion
                right = self.right_child(nid)
```

File: demo/json-model/json_parser.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                stacks.append(left)
+                stacks.append(rigth)
```
Comment (hcho3) [inline demo/json-model/json_parser.py]:
```suggestion
                stacks.append(right)
```

File: tests/python/test_basic_models.py [hunk 3]
```diff
@@ -11,6 +13,20 @@
 rng = np.random.RandomState(1994)
 
 
+def json_model():
```
[the remark this answers is not in the document]
Comment (trivialfis) [reply tests/python/test_basic_models.py]: @hcho3 Done.

**Pull request merged**